src/PollingBlockTracker.ts

```diff
@@ -272,14 +271,7 @@ export class PollingBlockTracker
     }
 
     log('Making request', req);
-    const res = await pify((cb) => this._provider.sendAsync(req, cb))();
-    log('Got response', res);
-    if (res.error) {
-      throw new Error(
-        `PollingBlockTracker - encountered error fetching block:\n${res.error.message}`,
-      );
-    }
-    return res.result;
+    return this._provider.request(req);
```

Comment:
Do we want to keep the log statements and error handling? We could use [`wrapError` from `@metamask/utils`](https://github.com/MetaMask/utils/blob/654f9cf4c2f64a5c83a4e9a42264bf0382409570/src/errors.ts#L90), which automatically preserves the entirety of original error (something that the previous code was not doing — it was just preserving the message but not the stack):

``` suggestion
    try {
      const result = this._provider.request(req);
      log('Got result', result);
      return result;
    } catch (error) {
      throw wrapError(error, 'PollingBlockTracker - encountered error fetching block');
    }
```

Reply:
Yes, I see what you mean. I guess since `request` now automatically throws if the response has an error we do not need to throw an error ourselves.

I think this is useful from a debugging perspective though, so what are your thoughts on adding a log statement at least then? Perhaps something like this (also, you can supply type parameters to `request` instead of having to typecast):

``` suggestion
    try {
      const result = await this._provider.request<[], string>(req);
      log('Got result', result);
      return result;
    } catch (error) {
      log('Encountered error fetching block', getErrorMessage(error));
      throw error;
    }
```